### PR TITLE
Multiplayer Match Diff support

### PR DESCRIPTION
**Pull request by mofirouz**

### CODE REVIEW

File: Classes/HLSessionClient.h [hunk 1]
```diff
@@ -176,6 +183,16 @@
 -(PMKPromise*)submitTurn:(NSNumber*)turn withData:(NSString*)data toNextGamer:(NSString*)nextGamerNickname forMatchWithId:(NSString*)matchId;
```
Comment (zyro) [inline Classes/HLSessionClient.h]: This should be marked as deprecated.

File: Classes/HLMatch.h [hunk 2]
```diff
@@ -32,18 +32,30 @@
...
 @property(readonly) NSNumber* turnCount;
 
 /** Name of gamer for the given turn */
 @property(readonly) NSString* turn;
 
+/** Gamer ID for the given turn */
+@property(readonly) NSString* turnGamerId;
+
 /** Nickname of all the gamers in the current match */
 @property(readonly) NSArray<NSString*>* gamers;
```
Comment (zyro) [inline Classes/HLMatch.h]: This should be marked as deprecated.

File: Classes/HLMatch.h [hunk 3]
```diff
@@ -32,18 +32,30 @@
...
 @property(readonly) NSString* matchId;
 
+/** Match Filters */
+@property(readonly) NSArray<NSString*>* filters;
+
 /** Current turn number */
 @property(readonly) NSNumber* turnCount;
 
 /** Name of gamer for the given turn */
 @property(readonly) NSString* turn;
```
Comment (zyro) [inline Classes/HLMatch.h]: This should be marked as deprecated.

**Comment by mofirouz** [comment]
@zyro Marked the method and fields as deprecated.

**Comment by zyro** [comment]
:+1:

**Pull request closed without merging**